flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/workflow/scheduler/EmbeddedQuartzScheduler.java

```diff
@@ -214,16 +248,230 @@ private void checkJobExists(JobKey jobKey, String errorMsg)
-    private class EmbeddedSchedulerJob implements Job {
+    public static class EmbeddedSchedulerJob implements Job {
+
+        public EmbeddedSchedulerJob() {}
 
         @Override
         public void execute(JobExecutionContext context) throws JobExecutionException {
-            JobDataMap dataMap = context.getJobDetail().getJobDataMap();
-            String workflowJsonStr = dataMap.getString(WORKFLOW_INFO);
-            WorkflowInfo workflowInfo = fromJson(workflowJsonStr, WorkflowInfo.class);
-            // TODO: implement the refresh operation for materialized table, see FLINK-35348
-            LOG.info("Execute refresh operation for workflow: {}.", workflowInfo);
+
+            SessionHandle sessionHandle = null;
+            OperationHandle operationHandle = null;
+            SqlGatewayRestClient gatewayRestClient = null;
+            try {
+                JobDataMap dataMap = context.getJobDetail().getJobDataMap();
+                String workflowJsonStr = dataMap.getString(WORKFLOW_INFO);
+                WorkflowInfo workflowInfo = fromJson(workflowJsonStr, WorkflowInfo.class);
+                LOG.info("Execute refresh operation for workflow: {}.", workflowInfo);
+
+                Configuration configuration = new Configuration();
+                RestClient restClient =
+                        RestClient.forUrl(
+                                configuration,
+                                Executors.newFixedThreadPool(1),
+                                new URL(workflowInfo.getRestEndpointUrl()));
+
+                String schedulerTime = dateToString(context.getScheduledFireTime());
+
+                gatewayRestClient = new SqlGatewayRestClient(workflowInfo.getRestEndpointUrl());
+                sessionHandle =
+                        gatewayRestClient.openSession(
+                                String.format(
+                                        "%s-quartz-refresh-session-%s",
+                                        workflowInfo.getMaterializedTableIdentifier(),
+                                        schedulerTime));
+                operationHandle =
+                        gatewayRestClient.refreshMaterializedTable(
+                                sessionHandle,
+                                workflowInfo.getMaterializedTableIdentifier(),
+                                true,
+                                schedulerTime,
+                                workflowInfo.getDynamicOptions(),
+                                Collections.emptyMap(),
+                                workflowInfo.getExecutionConfig());
+
+                List<RowData> results =
+                        gatewayRestClient.fetchOperationAllResults(sessionHandle, operationHandle);
+
+                String jobId = results.get(0).getString(0).toString();
+                LOG.info(
+                        "Successfully executed refresh operation for materialized table: {} with job id: {}.",
```

Comment:
`Successfully execute refresh operation for materialized table: {} with job id: {}.` ?